src/shred_network/repair_service.zig

```diff
@@ -251,7 +252,8 @@ pub const RepairRequester = struct {
     logger: ScopedLogger(@typeName(Self)),
     random: Random,
     keypair: *const KeyPair,
-    sender: SocketThread,
+    sender_pipe: *SocketPipe,
+    sender_channel: *Channel(Packet),
```

Comment:
it almost feels like the channel should belong (or be initialized) inside the pipe -- wdyt

Reply:
The advantage of decoupling them is that it's less opinionated about the relationship between channels and pipes. For example you may want multiple sockets using the same channel.

Reply:
This was the use-case of `SocketThread`. But of the 4 places using SocketPipe, there's one where it cant couple them:

`GossipService` can be initialized separately & optionally creates the pipe (among other things) in `.start()`. Some tests dont call start() and instead mimic a receiver pipe by pushing data directly to the channels themselves (see `test packet verification` and `benchmarkGossipService`).